include/numerics/petsc_vector.h

```diff
@@ -349,7 +349,7 @@ class PetscVector final : public NumericVector<T>
   virtual std::unique_ptr<NumericVector<T>>
   get_subvector(const std::vector<numeric_index_type> & rows) override;
 
-  virtual void restore_subvector(NumericVector<T> && subvector,
+  virtual void restore_subvector(std::unique_ptr<NumericVector<T>> && subvector,
```

Comment:
I would just pass the `std::unique_ptr` by value instead of rvalue reference. This is what we do in `MeshBase::add_elem()` for example, and it [guarantees that the function takes ownership of the pointer](https://stackoverflow.com/questions/71930850/why-not-to-accept-stdunique-ptr-by-rvalue-reference).